### PR TITLE
Create & Save Shopcart

### PR DESCRIPTION
**Pull request by radhikamattoo**
### Overview
* Added `save()` method and class-level `remove_all()` and `all()` methods to `shopcart.py`
* Added `test_it_can_be_saved()` and `setUp()` methods to `test_shopcart.py` 

### Running
After checking out the `create_shopcart` branch: 
`vagrant ssh`
`cd /vagrant`
`nosetests`

All 4 test cases should pass.

### CODE REVIEW

**Comment by thoughts1053** [comment]
@radhikamattoo I looked through the code and ran the tests locally and everything looked good. I just had a couple questions. I know this is modeled after the prof's repo so we're following suite but:

1. You're doing more than just implementing create. You've also implemented get all and delete all which are separate issues.
2. We still have 100% code coverage, but the `all` and `remove_all` methods aren't being tested
3. I think we should make the `products` variable default to a list if `none` is passed in, so we can create a new shopcart by doing `Shopcart(1)`.

Just some thoughts. We can just keep everything as is as it looks good, follows the prof's setup, and still has 100% coverage.

**Comment by radhikamattoo** [comment]
@thoughts1053 
I totally agree - I tried to avoid implementing the class-level methods, but when it came to testing it PyUnit basically refused to let me access class-level properties directly.

My original code looked like this: 
```
def test_it_can_be_saved(self):
        """ Test Model is Saved to Memory """        
        cart = Shopcart(1,[])
        cart.save()
        self.assertEqual(len(Shopcart.__data),1)
```
Yet I would get the same error every time: 

`AttributeError: type object 'Shopcart' has no attribute '_TestShopcart__data'`

which is strange in and of itself, as PyUnit seems to just prepend `_TestShopcart` to the property I'm trying to access. 

I tried working around it for a while, but then I looked at the Professor's unit test for his save method, and decided to copy it. 

As for unit tests for `remove_all` and `all`, I'm not sure if I should implement that now, or leave that for the other issues you mentioned that handle `remove_all` and `all`? @carpetri @RegulusXiang What do you guys think? 

And wrt. your #3, I agree - I can add that in for this PR, just request changes and I'll commit it!

**Comment by thoughts1053** [comment]
@radhikamattoo you can just add another commit and push the branch. No need to request changes.

**Comment by radhikamattoo** [comment]
@thoughts1053 done!

File: models/shopcart.py [hunk 1]
```diff
@@ -17,11 +17,25 @@ class Shopcart(object):
 
     __products = dict()
 
-    def __init__(self, uid=0, products=None):
+    def __init__(self, uid=0, products=[]):
```
Comment (thoughts1053) [inline models/shopcart.py]: You should never have a list (immutable object) as a default argument in python cause it has some funky default behavior: http://docs.python-guide.org/en/latest/writing/gotchas/.

A better alternative is leave `None` as the default argument and then in the body do something like `self.products = products if products else []`.

That's a one-liner, but you can also break it out into a regular if/else statement.

Comment (carpetri) [reply models/shopcart.py]: Nice to know. Thanks @thoughts1053

**Comment by carpetri** [comment]
So I are we leaving the extra things in this issue or not? @radhikamattoo @thoughts1053 @RegulusXiang

**Comment by thoughts1053** [comment]
I'm fine leaving it as is for now. We're following the prof's setup anyway.

**Pull request merged**